docs/csharp/tutorials/attributes.md

```diff
@@ -0,0 +1,164 @@
+
+Attributes provide a way of associating information with code in a declarative way. In this tutorial,
+you'll be introduced to how to add attributes to your code, how to create and use your own attributes, and how to use some
+attributes that are built into .NET Core.
+
+## Prerequisites
+You’ll need to setup your machine to run .NET core. You can find the
+installation instructions on the [.NET Core](https://www.microsoft.com/net/core)
+page.
+You can run this application on Windows, Ubuntu Linux, macOS or in a Docker container. 
+You’ll need to install your favorite code editor. The descriptions below
+use [Visual Studio Code](https://code.visualstudio.com/) which is an open
+source, cross platform editor. However, you can use whatever tools you are
+comfortable with.
+
+## Create the Application
+
+Now that you've installed all the tools, create a new .NET Core
+application. To use the command line generator, execute the following command in your favorite shell:
+
+`dotnet new`
+
+This command will create a barebones .NET core project with `project.json` and `Program.cs` files. You will need to execute `dotnet restore` to restore the dependencies needed to compile this project.
+
+To execute the program, use `dotnet run`. You should see "Hello, World" output to the console.
+
+## How to add attributes to code
+
+In C#, attributes are classes that inherit from the `Attribute` base class. Any class that inherits from `Attribute` can be used as a sort of "tag" on other pieces of code.
+For instance, there is an attribute called `ObsoleteAttribute`. This is used to signal that code is obsolete and shouldn't be used anymore. You can place this attribute on a class,
+for instance, by using square brackets.
+
+[!code-csharp[Obsolete attribute example](../../../samples/snippets/csharp/tutorials/attributes/Program.cs#ObsoleteExample1)]  
+
+Note that while the class is called `ObsoleteAttribute`, it's only necessary to use `[Obsolete]` in the code. This is a convention that C# follows.
+You can use the full name `[ObsoleteAttribute]` if you choose.
+
+When marking a class obsolete, it's a good idea to provide some information as to *why* it's obsolete, and/or *what* to use instead. Do this by passing a string
+parameter to the Obsolete attribute.
+
+[!code-csharp[Obsolete attribute example with parameters](../../../samples/snippets/csharp/tutorials/attributes/Program.cs#ObsoleteExample2)]
+
+The string is being passed as an argument to an `ObsoleteAttributes` constructor, just as if you were writing `var attr = new ObsoleteAttribute("some string")`.
+
+Parameters to an attribute constructor are limited to simple types/literals: `bool, int, double, string, Type, enums, etc` and arrays of those types.
+You can not use an expression or a variable. You are free to use positional or named parameters.
+_There is a 'gotcha' here to watch out for: the C# compiler won't stop you from creating those parameters, it will only stop you from using them_.
```

Comment:
This begs for an example. 

When we include examples that don't work at compile time, we include them inline:

```csharp
int j = 400;
byte b = j; // does not compile
```

We do this so broken code doesn't break the CI build.